api/document/models.py

```diff
@@ -50,6 +50,17 @@ def annotations(self) -> Iterator['Annotation']:
                                self.inlinerequirements.all())
 
 
+def annotate_with_has_docnodes(queryset=None):
+    """We frequently want to filter a Policy queryset by whether or not the
+    policies have associated DocNodes. This annotates a queryset with that
+    data."""
+    if queryset is None:
+        queryset = Policy.objects.all()
+
+    subquery = DocNode.objects.filter(policy=models.OuterRef('pk'))
+    return queryset.annotate(has_docnodes=models.Exists(subquery))
```

Comment:
Have you considered making this a [custom `QuerySet` method](https://docs.djangoproject.com/en/2.0/topics/db/managers/#calling-custom-queryset-methods-from-the-manager) instead of a function?  You are definitely way more experienced with Django than me so I'll leave it up to you... I assume you have already considered this option and decided it was horrible for some reason, though I am curious to know why.

Reply:
It's a great idea, but I was avoiding it due to circular imports. Documents reference Policies; adding a method to Policies (indirectly through custom manager) referencing Documents would make the loop. There might be different way to write the subquery, though. I'll poke around a bit more.